src/process/macos.rs

```diff
@@ -298,3 +299,136 @@ fn clone_task_all_info(src: &TaskAllInfo) -> TaskAllInfo {
+    p_back: vm_types::user_addr_t,
+}
+
+#[repr(C)]
+#[derive(Copy, Clone)]
+pub union p_un {
+    pub p_st1: run_sleep_queue,
+    pub p_starttime: libc::timeval,
+}
+
+#[repr(C)]
+#[derive(Copy, Clone)]
+pub struct extern_proc {
+    pub p_un: p_un,
+    pub p_vmspace: vm_types::user_addr_t,
+    pub p_sigacts: vm_types::user_addr_t,
+
+    pub p_flag: libc::c_int,
+    pub p_stat: libc::c_char,
+    pub p_pid: libc::pid_t,
+    pub p_oppid: libc::pid_t,
+    pub p_dupfd: libc::c_int,
+    pub user_stack: caddr_t,
+    pub exit_thread: *mut libc::c_void,
+    pub p_debugger: libc::c_int,
+    pub sigwait: boolean::boolean_t,
+    pub p_estcpu: libc::c_uint,
+    pub p_cpticks: libc::c_int,
+    pub p_pctcpu: u32,
+    pub p_wchan: *mut libc::c_void,
+    pub p_wmesg: *mut libc::c_char,
+    pub p_swtime: libc::c_uint,
+    pub p_slptime: libc::c_uint,
+    pub p_realtimer: libc::itimerval,
+    pub p_rtime: libc::timeval,
+    pub p_uticks: u64,
+    pub p_sticks: u64,
+    pub p_iticks: u64,
+    pub p_traceflag: libc::c_int,
+    pub p_tracep: *mut libc::c_void,
+    pub p_siglist: libc::c_int,
+    // TODO: It was a pointer to `struct vnode`
+    pub p_textvp: *mut libc::c_void,
+    pub p_holdcnt: libc::c_int,
+    pub p_sigmask: libc::sigset_t,
+    pub p_sigignore: libc::sigset_t,
+    pub p_sigcatch: libc::sigset_t,
+    pub p_priority: libc::c_uchar,
+    pub p_usrpri: libc::c_uchar,
+    pub p_nice: libc::c_char,
+    pub p_comm: [libc::c_char; 17],
+    // TODO: It was a pointer to `struct proc`, declared at `bsd/sys/proc.h`
+    pub p_pgrp: *mut libc::c_void,
+    // TODO: It was a pointer to `struct user`, declared at `bsd/sys/user.h`
+    // but it is not used anymore and we do not need it too
+    pub p_addr: *mut libc::c_void,
+    pub p_xstat: libc::c_ushort,
+    pub p_acflag: libc::c_ushort,
+    pub p_ru: *mut libc::rusage,
+}
+
+#[repr(C)]
+#[derive(Copy, Clone)]
+pub struct kinfo_proc_eproc {
+    // TODO: It should be a pointer to `struct proc`
+    pub e_paddr: *mut libc::c_void,
+    // TODO: It should be a pointer to `struct session`
+    // but since we are not using it and it's declaration kinda big,
+    // it was skipped. Same goes to `e_tsess` field below.
+    pub e_sess: *mut libc::c_void,
+    pub e_pcred: pcred,
+    pub e_ucred: libc::xucred,
+    pub e_vm: vmspace,
+    pub e_ppid: libc::pid_t,
+    pub e_pgid: libc::pid_t,
+    pub e_jobc: libc::c_short,
+    pub e_tdev: libc::dev_t,
+    pub e_tpgid: libc::pid_t,
+    pub e_tsess: *mut libc::c_void, // TODO: See `TODO` comment from above
+    pub e_wmesg: [libc::c_char; 8],
+    pub e_xsize: segsz_t,
+    pub e_xrssize: libc::c_short,
+    pub e_xccount: libc::c_short,
+    pub e_xswrss: libc::c_short,
+    pub e_flag: i32,
+    pub e_login: [libc::c_char; 12],
+    pub e_spare: [i32; 4],
+}
+
+pub const CTL_MAXNAME: i32 = 12;
+pub const P_TRANSLATED: i32 = 131072;
+pub const CPU_TYPE_X86_64: i32 = 16777223;
+pub const CPU_TYPE_ARM64: i32 = 16777228;
```

Comment:
If the above definitions are used by `arch.rs` only, it may be better that they are placed in `arch.rs`.